File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.js
function solution(n) {
    var answer = 0;
    let arr = [0, 1]
    
    for (let i = 2; i < n+1; i++) {
        arr.push(arr[i-1]%1234567 + arr[i-2]%1234567)
     
    }
    answer = arr[n]%1234567

    
    return answer;
}

File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.java
class Solution {
    static long MOD = 1234567;
    static long[] memo = new long[100001];
    
    static long fibo(int n) {
        if (memo[n] > 0) return memo[n];
        
        if (n == 0) return 0;
        if (n == 1) return 1;
        
        long res = (fibo(n-1) % MOD + fibo(n-2) % MOD) % MOD;
        memo[n] = res;
        return res;
    }
    
    public long solution(int n) {
        return fibo(n) % MOD;
    }
}